jme3-core/src/main/java/com/jme3/anim/tween/action/BlendAction.java

```diff
@@ -1,129 +1,265 @@
-        firstActiveAction.setCollectTransformDelegate(this);
-        secondActiveAction.setCollectTransformDelegate(this);
-
-        //only interpolate the first action if the weight if below 1.
+    public void doInterpolate(final double t) {
+        final float blendWeight = blendSpace.getWeight();
+        final BlendableAction firstActiveAction = (BlendableAction) actions[firstActiveIndex];
+        final BlendableAction secondActiveAction = (BlendableAction) actions[secondActiveIndex];
+        //activate the collection delegation -> will skip the transitionWeight value calculation.
+        activateTransformationDelegation(firstActiveAction, secondActiveAction);
+        //only interpolate the first action if the weight is below 1.
         if (blendWeight < 1f) {
-            firstActiveAction.setWeight(1f);
-            firstActiveAction.interpolate(t * timeFactor[firstActiveIndex]);
-            if (blendWeight == 0) {
-                for (HasLocalTransform target : targetMap.keySet()) {
-                    collect(target, targetMap.get(target));
-                }
-            }
+            interpolateAction(firstActiveAction, 1f, t * timeFactor[firstActiveIndex]);
         }
+        //protect the second action from extrapolation & negative values.
+        interpolateAction(secondActiveAction, Math.min(Math.max(blendWeight, 0f), 1f), t * timeFactor[secondActiveIndex]);
+        //release the resources.
+        resetTransformationDelegation(firstActiveAction, secondActiveAction);
+    }
 
-        //Second action should be interpolated
-        secondActiveAction.setWeight(blendWeight);
-        secondActiveAction.interpolate(t * timeFactor[secondActiveIndex]);
+    /**
+     * Resets the delegation & release the resources.
+     * @param firstAction the firstActiveAction to release.
+     * @param secondAction the secondActiveAction to release.
+     */
+    private void resetTransformationDelegation(final BlendableAction firstAction, final BlendableAction secondAction){
+        firstAction.setCollectTransformDelegate(null);
+        secondAction.setCollectTransformDelegate(null);
+    }
 
-        firstActiveAction.setCollectTransformDelegate(null);
-        secondActiveAction.setCollectTransformDelegate(null);
+    /**
+     * Activates the delegation to run this factory method #{@link BlendAction#collectTransform(HasLocalTransform, Transform, float, BlendableAction)}
+     * from the firstAction & the secondAction #{@link ClipAction#doInterpolate(double)} method for interpolation.
+     * @param firstAction the firstActiveAction.
+     * @param secondAction the secondActiveAction.
+     */
+    private void activateTransformationDelegation(final BlendableAction firstAction, final BlendableAction secondAction){
+        firstAction.setCollectTransformDelegate(this);
+        secondAction.setCollectTransformDelegate(this);
+    }
 
+    /**
+     * Interpolates the actions according to a blendingWeight(delta value).
```

Comment:
The Greek letter "delta" usually stands for change. The blend weight is neither a delta value nor a delta factor.

Reply:
The blending weight is a changeable value through the `BlendSpace` either Linear or non-Linear equations can change it.

Reply:
Just because a value changes over time does not make it a delta.

Reply:
Alright, i would remove the keyword 'delta'.